de.metas.business/src/main/java/org/adempiere/pricing/api/IMDiscountSchemaBL.java

```diff
@@ -48,7 +52,7 @@
 	 * @param bPartnerFlatDiscount
 	 * @return
 	 */
-	BigDecimal calculateDiscount(I_M_DiscountSchema schema, BigDecimal Qty, BigDecimal Price, int M_Product_ID, int M_Product_Category_ID, BigDecimal bPartnerFlatDiscount);
+	DiscountRequest calculateDiscount(I_M_DiscountSchema schema, BigDecimal Qty, BigDecimal Price, int M_Product_ID, int M_Product_Category_ID, BigDecimal bPartnerFlatDiscount);
```

Comment:
as you used it, i this class is a DiscountResult .

More, you could group the method params and create the DiscountRequest.
You will be able to unify all calculateDiscount methods using DiscountRequest.